crates/core/src/id.rs

```diff
@@ -16,6 +16,33 @@ pub(super) mod constants {
     pub(super) const HEX_LEN: usize = LEN * 2;
 }
 
+#[macro_export]
+/// Generate newtypes for `Id`s identifying Repository files
+macro_rules! new_id {
```

Comment:
Would name it `define_new_id_struct`, so it's clear, here is a new struct being defined.

Reply:
renamed.